ondine/homework2.md

```diff
@@ -0,0 +1,246 @@
+### Ondine Gallatin
+
+### Homework #2
+
+Due Monday, 1/19
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** _(Easy, 10% of total time)_
+
+**a)** Write a function `fractionString(n,d)` which takes 2 parameters (n,d) and returns a string.  For example, 
+`fractionString(7,4)` should return "1 3/4", and `fractionString(3,3)` should (for now) return "1 0/3".  As before, assume that _n_ and _d_ are both positive integers.
+
+```
+function fractionString(n, d) {
+  var numeratorTwo = n % d;
+  var number = Math.floor(n / d);
+  var result = number + ' ' + numeratorTwo + '/' + d;
+  return result;
+}
+```
+
+**b)** Write a second version of your function which improves the output in two special cases:
+
+- An output string like "0 1/2" should be simplified to just "1/2";
+- An output string like "1 0/3" should be simplified to just "1".
+
+```
+function fractionStringTwo(n, d) {
+  var numeratorTwo = n % d;
+  var number = Math.floor(n / d);
+  if (number < 1) {
+    var result = numeratorTwo + '/' + d;
+  } else if (numeratorTwo == 0) {
+    result = number;
+  } else {
+    result = number + ' ' + numeratorTwo + '/' + d;
+    };
+  return result;
+}
+```
+---
+
+**2)** _(Moderate, 15%)_
+
```

Comment:
In this section, it is often useful to break the components out line by line. I often do this if I am deciphering someone's code at work.  I will literally take each section of the code (often just one or two lines). Paste it into the console. and play with it. What if this variable is true, false, a number. How does it react?

Then I will proceed through the function until I have pieced out each component. 

Then I will put it all together and pass it different variables, predicting the result. It's not until I am accurately predicting the result that I can proceed with refactoring the function to make more sense. 

When you're starting coders often just try to read foreign code like a book. It never works that way. you have to take the time to DIGEST the information. For an example, before I graded this assignment I spent about 10 min each just chopping the original functions apart. That's not including taking the time to refactor.